src/python/pants/rules/core/cloc.py

```diff
@@ -0,0 +1,119 @@
+  digest = Digest(sha_256, 546279)
+  snapshot = yield Get(Snapshot, UrlToFetch(url, digest))
+  yield DownloadedClocScript(script_path=snapshot.files[0], digest=snapshot.directory_digest)
+
+
+class CountLinesOfCode(LineOriented, Goal):
+  name = 'cloc'
+
+  @classmethod
+  def register_options(cls, register) -> None:
+    super().register_options(register)
+    register('--transitive', type=bool, fingerprint=True, default=True,
+             help='Operate on the transitive dependencies of the specified targets.  '
+                  'Unset to operate only on the specified targets.')
+    register('--ignored', type=bool, fingerprint=True,
+             help='Show information about files ignored by cloc.')
+
+
+@console_rule(CountLinesOfCode, [Console, CountLinesOfCode.Options, DownloadedClocScript, Specs])
+def run_cloc(console, options, cloc_script, specs):
+  """Runs the cloc perl script in an isolated process"""
+
+  transitive = options.values.transitive
+  ignored = options.values.ignored
+
+  if (transitive):
+    targets = yield Get(TransitiveHydratedTargets, Specs, specs)
+    all_target_adaptors = {t.adaptor for t in targets.closure}
+  else:
+    targets = yield Get(HydratedTargets, Specs, specs)
+    all_target_adaptors = {t.adaptor for t in targets}
+
+  build_root = get_buildroot()
+  source_paths: Set[str] = set()
+  for t in all_target_adaptors:
+    sources = getattr(t, 'sources', None)
+    if sources is not None:
+      for f in sources.snapshot.files:
+        source_paths.add(str(Path(build_root, f)))
+
+  file_content = '\n'.join(sorted(source_paths)).encode()
+
+  input_files_filename = 'input_files.txt'
+  report_filename = 'report.txt'
+  ignore_filename = 'ignored.txt'
+
+  input_file_list = InputFileContent(path=input_files_filename, content=file_content)
+  input_file_digest = yield Get(Digest, InputFileContent, input_file_list)
+  cloc_script_digest = cloc_script.digest
+  digest = yield Get(Digest, DirectoriesToMerge(directories=(cloc_script_digest, input_file_digest)))
+
+  cmd = (
+    '/usr/bin/perl',
+    cloc_script.script_path,
+    '--skip-uniqueness', # Skip the file uniqueness check.
+    f'--ignored={ignore_filename}', # Write the names and reasons of ignored files to this file.
+    f'--report-file={report_filename}', # Write the output to this file rather than stdout.
+    f'--list-file={input_files_filename}', # Read an exhaustive list of files to process from this file.
+  )
+
+  req = ExecuteProcessRequest(
+    argv=cmd,
+    input_files=digest,
+    output_files=(report_filename, ignore_filename),
+    description='cloc',
+  )
+
+  exec_result = yield Get(ExecuteProcessResult, ExecuteProcessRequest, req)
+  files_content = yield Get(FilesContent, Digest, exec_result.output_directory_digest)
+
+  file_outputs = {fc.path: fc.content.decode() for fc in files_content.dependencies}
+
+  with CountLinesOfCode.line_oriented(options, console) as (print_stdout, print_stderr):
```

Comment:
This goal isn't line oriented I don't think: the mixin adds things that would break the output of cloc (separator character).

I think you want to skip extending LineOriented, and just use the methods of Console directly.